### PR TITLE
Explore: Consolidate Explore links and generic data links functions

### PR DESCRIPTION
**Pull request by gelicia**
This is a POC/WIP to coordinate with @ifrost. This PR will be closed when we are ready for review.

---

**What is this feature?**

[Add a brief description of what the feature or update does.]

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by torkelo — COMMENTED** [review]
Please hold on merging this, need to review this. At a glance feels like too much explore specific things creeping into really core bits and we might need a big refactor around datalinks to make it more flexible.

Also the new dashboard architecture and templating engine might influence how this could work

File: packages/grafana-data/src/field/fieldOverrides.ts [hunk 1]
```diff
@@ -201,13 +203,13 @@ export function applyFieldOverrides(options: ApplyFieldOverrideOptions): DataFra
...
-        newFrame,
-        field,
-        field.state!.scopedVars,
-        context.replaceVariables,
-        options.timeZone
-      );
+      field.getLinks = getLinksSupplier({
+        frame: newFrame,
+        field: field,
+        fieldScopedVars: field.state!.scopedVars,
```
Comment (torkelo) [inline packages/grafana-data/src/field/fieldOverrides.ts:209]: Why are we passing this when the function is already getting passed the field

**Comment by torkelo** [comment]
Please try to add more detail in PR body on the problems and features this PR tries to solve

**Comment by gelicia** [comment]
> Please try to add more detail in PR body on the problems and features this PR tries to solve

This is a draft PR, meant to coordinate work with another team member. I would, of course, add more details when the PR was ready to be reviewed.

File: packages/grafana-data/src/field/fieldOverrides.ts (hunk 1 above)
Comment (gelicia) [reply packages/grafana-data/src/field/fieldOverrides.ts:209]: I can look into this but my main concern with this was finding out if we could combine two very similar functions without regressions. This is not a change in logic compared to what's there now - the only thing changed was having the params be a single object.

**Comment by ifrost** [comment]
It looks like this will mainly consolidate the different patterns we currently use so it shouldn't introduce more technical debt. I agree it needs further improvements, for example, `linkSupplier` shouldn't depend on Explore-specific logic (though it currently does as it generates Explore URL). Putting everything through one function could help refactor it further.

**Pull request closed without merging**